shaq/_cli.py

```diff
@@ -164,11 +164,74 @@ def _parser() -> argparse.ArgumentParser:
         type=str,
         help="send the request to a proxy server",
     )
+    advanced_group.add_argument(
+        "--metadata",
+        action="store_true",
+        help="Overwrite titel and metadata of the song",
+    )
     return parser
 
+def updateMetadata(console, title, artist, path):
+    
+    # Imports happen here so that when this functionality is not used, mutagen is not needed to run the program
+    # This is to avoid unnecessary dependencies.
+
+    try:
+        from mutagen.flac import FLAC
+        from mutagen.mp3 import MP3
+        from mutagen.id3 import TPE2, TIT2, TPE1
+        from mutagen.oggvorbis import OggVorbis
+    except ImportError:
+        console.print("[red]mutagen library is not installed. Install it with 'pip install mutagen'[/red]")
+        sys.exit(1)
```

Comment:
Could you please declare this dependency in the `pyproject.toml` instead? If it's optional, it should be put inside a optional dependency (i.e. "extra"). That way we'll still need to check for it at import time, but at least the version will be managed consistently (and dependency update tooling will track it correctly).

See here for more details: <https://packaging.python.org/en/latest/specifications/pyproject-toml/#dependencies-optional-dependencies>